### PR TITLE
Bugs solved

### PR DESCRIPTION
**Pull request by ezequielsetzesglobant**
Two bugs were resolved, the internet check in the form and the filter and value messages in the search dialog of the screen with the list of assets.

Video bug 1:
https://user-images.githubusercontent.com/84338513/152015397-354ada10-6b4e-4384-a54e-4969b61fd6b2.mov

Video bug 2:
https://user-images.githubusercontent.com/84338513/151814464-81e6bd9b-e883-4dc1-8825-a6f1aad27628.mov

### CODE REVIEW

File: stockApp/stockApp/Presentation/View/FormViewController.swift [hunk 1]
```diff
@@ -4,83 +4,101 @@ import UIKit
...
+    var reachability = try? Reachability()
+
+    // MARK: - Setup
+    init(viewModel: FormViewModelProtocol,
+         isFromResultsList: Bool,
+         helper: HelperFunctions,
+         coordinator: CoordinatorProtocol) {
+        self.viewModel = viewModel
+        self.isFromResultsList = isFromResultsList
+		self.alertHelper = helper
```
Comment (ezequielvelezglobant) [inline stockApp/stockApp/Presentation/View/FormViewController.swift:36]: fix alignment

File: stockApp/stockApp/Presentation/View/FormViewController.swift [hunk 2]
```diff
@@ -4,83 +4,101 @@ import UIKit
...
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setUpViewModel()
+        setUpUI()
 		if isFromResultsList {
 			superView.barcodeUITextField.isEnabled = false
 		} else {
 			viewModel.checkAutomaticBarcode(view: superView)
 		}
```
Comment (ezequielvelezglobant) [inline stockApp/stockApp/Presentation/View/FormViewController.swift:53]: fix alignment

File: stockApp/stockApp/Presentation/View/FormViewController.swift [hunk 3]
```diff
@@ -4,83 +4,101 @@ import UIKit
...
+            var isNotConnectionInternet = false
+            self.reachability = try? Reachability()
+            self.reachability?.whenUnreachable = { _ in
+                self.displayisNotConnectedToInternetAlert()
+                isNotConnectionInternet = true
+            }
+            do {
+                try self.reachability?.startNotifier()
+            } catch {
+            }
```
Comment (ezequielvelezglobant) [inline stockApp/stockApp/Presentation/View/FormViewController.swift:76]: catch errors.

File: stockApp/stockApp/Presentation/ViewModel/SearchViewModel.swift [hunk 4]
```diff
@@ -29,5 +33,6 @@ final class SearchViewModel: SearchViewModelProtocol {
 
     struct Constants {
         static let emptyValueError = "Please insert a value"
+        static let emptyFilterError = "Please insert a filter"
```
Comment (ezequielvelezglobant) [inline stockApp/stockApp/Presentation/ViewModel/SearchViewModel.swift:36]: you "select" a filter

**Review by ezequielvelezglobant — CHANGES_REQUESTED** [review]

File: stockApp/stockApp/Presentation/View/AssetViewController.swift [hunk 5]
```diff
@@ -80,14 +80,14 @@ class AssetViewController: UIViewController {
...
 	func displayThereIsNotInternetMessage() {
-
 		helper.showAlertMessage(title: Strings.isNotConnectedToInternet, viewController: self)
 	}
 }
 
 extension AssetViewController: AssetViewDelegate {
 
 	func searchButtonPressed(from view: AssetView) {
+        searchAlertView.delegate = self
```
[the remark this answers is not in the document]
Comment (ezequielsetzesglobant) [reply stockApp/stockApp/Presentation/View/AssetViewController.swift:90]: done

File: stockApp/stockApp/Presentation/View/FormViewController.swift (hunk 1 above)
Comment (ezequielsetzesglobant) [reply stockApp/stockApp/Presentation/View/FormViewController.swift:36]: done

File: stockApp/stockApp/Presentation/View/FormViewController.swift (hunk 2 above)
Comment (ezequielsetzesglobant) [reply stockApp/stockApp/Presentation/View/FormViewController.swift:53]: done

File: stockApp/stockApp/Presentation/ViewModel/SearchViewModel.swift (hunk 4 above)
Comment (ezequielsetzesglobant) [reply stockApp/stockApp/Presentation/ViewModel/SearchViewModel.swift:36]: done

File: stockApp/stockApp/Presentation/View/FormViewController.swift (hunk 3 above)
Comment (ezequielsetzesglobant) [reply stockApp/stockApp/Presentation/View/FormViewController.swift:76]: done

**Review by ezequielvelezglobant — APPROVED** [review]

**Pull request merged**